src/trading/takeBidOrder.se

```diff
@@ -149,9 +146,9 @@ def takeBidOrder(taker: address, orderID: bytes32, market: address, outcome: uin
     # maker is opening a long, taker is opening a short
     if takerDesiredSharesForThisOrder:
         # transfer cash from both parties into this contract for complete set purchase
-        takerPortionOfCompleteSetCost = safeFxpMul(takerDesiredSharesForThisOrder, sharePriceShort)
+        takerPortionOfCompleteSetCost = safeDiv(safeMul(takerDesiredSharesForThisOrder, sharePriceShort), market.getMaxDisplayPrice() - market.getMinDisplayPrice())
         denominationToken.transferFrom(taker, self, takerPortionOfCompleteSetCost)
-        makerPortionOfCompleteSetCost = safeFxpMul(takerDesiredSharesForThisOrder, sharePriceLong)
```

Comment:
I think we can remove `sharePriceLong` from this function now.

Reply:
I'm working on migrating `takeBidOrder` to solidity right now, which still has `sharePriceLong` but it does remove some other pieces.  Recommend bringing this back up once that PR is out if you believe removal still makes sense (and why).